### DISCUSSION TITLE
Is it possible to restrict registrations to certain email domains but only for registration and not invitations ?

### DISCUSSION THREAD

**Discussion by codeurimpulsif** [dani-garcia/vaultwarden#4905] [Q&A]
Hi everyone,

I was wondering if it is possible to restrict global registrations to certain email domains but to not apply this restriction to invitations, specially invitations from organizations ?

For example, I would like to open registrations to anyone with an email at `example.com` only, but if an organization (created by someone with a `example.com` domain email) invite someone else with an `gmail.com` email account (or any other email domain without restriction) then this person can create an account as well.

Maybe something like `SIGNUPS_DOMAINS_WHITELIST` but called `INVITATIONS_DOMAINS_WHITELIST` where we can set `*` for "all email domains"?

Is it possible?

Thanks

**Comment by BlackDex** [comment]
I do not think that is something we would like to add. It will only make it more complex then it already is.
If you want to invite a user and bypass the restrictions use the `/admin` interface and later invite the user to the correct org's.

**Reply by codeurimpulsif** [reply]
In fact it's for users who are not global admin, but only admin of organizations :/

**Reply by BlackDex** [reply]
I understand that. But it will cause a lot of confusion having exception on exception and allowlist on allowlist.

It might be better to allow only specific org admin/owners to invite people, and allow all domains and no invites then.

You then must trust these owners and admins to invite the allowed people.

**Reply by BlackDex** [reply]
Or, just do not allow any signups, invite those people manually and let them create an org and then they can do whatever they want. You could then also add the specific user to the org invite users to have a better control on it.
